enhancements/ingress/proxy-protocol.md

```diff
@@ -0,0 +1,193 @@
+---
+title: proxy-protocol
+authors:
+  - "@Miciah"
+reviewers:
+  - "@candita"
+  - "@danehans"
+  - "@frobware"
+  - "@knobunc"
+  - "@miheer"
+  - "@rfredette"
+  - "@sgreene570"
+approvers:
+  - "@danehans"
+  - "@frobware"
+  - "@knobunc"
+creation-date: 2021-02-23
+last-updated: 2021-02-23
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Ingress PROXY Protocol
+
+This enhancement enables cluster administrators to enable [PROXY
+protocol](http://www.haproxy.org/download/2.2/doc/proxy-protocol.txt) on
+IngressControllers.
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement extends the IngressController API to allow cluster
+administrators to configure IngressControllers that use the "HostNetwork"
+and "NodePortService" endpoint publishing strategy types to use [PROXY
+protocol](http://www.haproxy.org/download/2.2/doc/proxy-protocol.txt).
+
+## Motivation
+
+When OpenShift is configured with an external load-balancer in front of an
+IngressController, the connections that the IngressController receives have the
+load balancer's address (instead of the original client addresses) as the source
+address.  PROXY protocol enables the external load-balancer to preserve the
+original client addresses and include them when forwarding connections to the
+IngressController, which can then use the original source addresses for
+filtering and injecting HTTP headers.
+
+### Goal
+
+1. Enable cluster administrators to enable PROXY protocol on IngressControllers that use the "HostNetwork" and "NodePortService" endpoint publishing strategy types.
+
+### Non-Goal
+
+1. Enabling PROXY protocol to be configured on individual Routes.
```

Comment:
Would a non-goal be allowing PROXY protocol to be configured when using the `LoadBalancerService` endpoint publishing strategy? (I guess mentioning this here would be slightly redundant but can't hurt to be explicit, considering our current usage of PROXY protocol on default ingress controllers on AWS)

Reply:
It would.  I added a non-goal as you suggested.  Thanks!